flask_resty/view.py

```diff
@@ -400,9 +401,14 @@ def request_args(self):
                 # KeyError for args that aren't present.
                 continue
 
-            value = args.getlist(field_name)
-            if not isinstance(field, fields.List) and len(value) == 1:
-                value = value[0]
+            value = args.get(field_name)
+            if isinstance(field, fields.List) and not hasattr(field, "delimiter"):
```

Comment:
maybe something like:

```
if (
    (
        isinstance(field, fields.List)
        or not isinstance(field, fields.DelimitedList)
    )
    and len(value) == 1
):
    value = value[0]
```

don't inline the deserialization bits from here; `DelimitedList` should take care of it.

Reply:
Sorry, could you clarify how this works? I thought we had to return a list when the argument is a `List`. Also, what do you mean by "don't inline the deserialization bits"?